provider/core/src/datagen/mod.rs

```diff
@@ -83,6 +83,14 @@ pub trait DataExporter: Sync {
     fn close(&mut self) -> Result<(), DataError> {
         Ok(())
     }
+
+    /// Returns the preferred fallback mode for this provider.
+    /// 
+    /// Providers capable of automatically including runtime fallback should return
+    /// [`FallbackMode::Full`]; all others should return [`FallbackMode::None`].
+    fn preferred_fallback_mode(&self) -> FallbackMode {
```

Comment:
So an issue I realized is that it could be a breakage if we add another fallback mode and then start passing that mode into `flush_with_built_in_fallback` just because this function returned `true`. I think it's best for this function to return a specific `BuiltInFallbackMode` which it also guarantees it supports in the flush function.

Reply:
I don't think that's an issue. A provider either supports builtin fallback or it doesn't. The non-exhaustiveness of the `RuntimeFallbackMode` takes care of exporters not knowing about future types, but I don't think supporting only `Naive` but not `Standard` makes sense.

Reply:
A data exporter probably implements something like

```rust
fn flush_with_built_in_fallback(&self, built_in_fallback_mode) -> Result {
    match built_in_fallback_mode {
        BuiltInFallbackMode::Standard => // do stuff
        _ => return DataError::custom("unknown fallback mode!")
    }
}
```

If we add `BuiltInFallbackMode::Naive` and then make it the preferred mode, that function starts failing _with the default icu_datagen::options::FallbackMode setting_ which is not good. That setting should always work. The way to guarantee that it works is to make the `preferred_built_in_fallback_mode()` function return something that the exporter knows it can supports, not just a `bool`.

Reply:
My scenario can happen for a 3P data exporter with our own datagen binary. We export the trait because we want people to write their own data exporters. I don't think this is a rare scenario.

Reply:
Oh yeah we allow byoexporter now. In any case, I'm happy to decree that `Standard` will be the default internal fallback mode forever, I don't think you want to make `Naive` the default either.

Reply:
Ok. changed to a bool